pkg/hubble/metrics/metrics.go

```diff
@@ -64,6 +86,18 @@ func initMetrics(address string, enabled api.Map, grpcMetrics *grpc_prometheus.S
 		errChan <- srv.ListenAndServe()
 	}()
 
+	go func() {
+		for pod := range podRemovalChan {
+			if pod.GetDeletionTimestamp() != nil {
```

Comment:
This blocks deletions of all other pod metrics. Instead we could probably use a proper queue type which lets us order things by when we want to delete them, rather than a channel. The `k8s.io/client-go/util/workqueue` is a good package for doing this that we already use in cilium.

Reply:
done